docs/adguard-browser-extension/mv3-version.md

```diff
@@ -0,0 +1,159 @@
+
+## Limitations
+
+### Maximum number of rules and filters
+
+Manifest V3 imposes limits on static and regex rules per extension. Users may receive notifications when these limits are exceeded.
+
+![Rule limits *border](https://cdn.adtidy.org/content/blog/new/rulelimits.png)
+
+![Too many extensions *border](https://cdn.adtidy.org/content/blog/new/other_extension.png)
+
+Basically, Manifest V3 divides rules into static (built-in) and dynamic, with strict limits in place.
+
+**Static rules:** minimum **30,000** rules per extension, with a cumulative limit of **330,000** for all extensions installed by a single user.
+
+For regex rules, the limit is set at **1,000 per extension**.
+
+The maximum number of simultaneously enabled filters is **50**.
+
+**Dynamic rules:** a strict cap of **5,000** rules is imposed, which includes a maximum of 1,000 regex rules.
+
+  If this limit is exceeded, only **5,000 converted rules** will be applied in the following order: first allowlist, then user rules, and finally — custom filters.
+
+> **Converted rules** are rules that have been transformed to [DNR](https://developer.chrome.com/docs/extensions/reference/api/declarativeNetRequest#build-rules) format using the [declarative converter](https://github.com/AdguardTeam/tsurlfilter/tree/master/packages/tsurlfilter/src/rules/declarative-converter). During this conversion process, some rules may overwrite others (`badfilter`), some may be combined (`removeparame`), resulting in a list of rules with a slightly different order.
+>
+> From this list of converted rules, we will only use 5,000 rules. The rest of them will be displayed in the editor, but not applied.
+
+ Here's how a rule with a basic modifier is converted to a declarative rule:
+
+  ```bash
+  ||example.org^$script,third-party,domain=example.com
+  ```
+
+  is converted to
+
+  ```bash
+  [
+  {
+    "id": 1,
+    "action": {
+    "type": "block"
+    },
+    "condition": {
+    "urlFilter": "||example.org^",
+    "domainType": "thirdParty",
+    "initiatorDomains": [
+      "example.com"
+    ],
+    "resourceTypes": [
+      "script"
+    ]
+    },
+    "priority": 302
+  }
+  ]
+  ```
+
+  More examples can be found on [GitHub](https://github.com/AdguardTeam/tsurlfilter/tree/release/v3.1/packages/tsurlfilter/src/rules/declarative-converter#basic-examples).
+
+### Network rules
+
+Network rules have limitations: some types of rules cannot be implemented in MV3, or can only be implemented with restrictions.
+
+Below is a list of modifiers that are supported, partially supported, or not supported at all. More details about the limitations can be found on [GitHub](https://github.com/AdguardTeam/tsurlfilter/tree/master/packages/tsurlfilter/src/rules/declarative-converter#readme). For further information on modifiers, please refer to our [Knowledge base](/general/ad-filtering/create-own-filters/#basic-rules-basic-modifiers).
+
+  | Basic modifiers | Supported | Not supported | Partially supported |
```

Comment:
_aged well_
https://uploads.adguard.org/sl797FA1l6hvbylFX6.png

\+ not sure about deprecated modifiers mentioning

Reply:
\+ I really do not like an idea of having one more compatibility table to maintain

Reply:
why not mention deprecated modifiers?
as for the compatibility table — I think it's up to you to decide since you are more likely to maintain it than the content team

Reply:
why not mention deprecated modifiers?
as for the compatibility table, I think it's up to you to decide, since you are more likely to maintain it than the content team

Reply:
I prefer to maintain compatibility tables in one place in the kb — in `create-own-filters.md`. and here we can only mention that some modifiers are not supported or have limitations, and `... check modifiers compatibility by <link>`

Reply:
I don't know why any table is needed `mv3-version.md`. I would remote it at all 🙈